### ISSUE TITLE
Didn't catch the reference in `ExportNamedDeclaration`

### ISSUE THREAD

**Issue by mysteryven** [oxc-project/oxc#3795]
The references of `A` is empty for:

```typescript
import { A } from 'foo'

export  { A }
```
![CleanShot 2024-06-20 at 19 39 38@2x](https://github.com/oxc-project/oxc/assets/33973865/300eeb26-9dc4-4dbb-9d55-b6c2f9c0991f)

[playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAIClgICAgICAgIC0G8rnONK89Iyo%2FIRvUICdKXoMpBKY%2Bxn64LMYJ7DwPsN%2Ff9OYgIA%3D)

**Comment by Boshen** [comment]
Related https://github.com/oxc-project/oxc/issues/3796

**Comment by Dunqing** [comment]
> Related #3796

This is not related tp #3796. `C` is not the identifier reference. So we cannot catch the reference

**Comment by Boshen** [comment]
It's related, it's not a `IdentifierReference`, but we need to create a reference for it.

Notice there are 2 references:

<img width="575" alt="image" src="https://github.com/oxc-project/oxc/assets/1430279/da473a0b-d886-4b71-a219-a0cc2b81ec82">


in binder.ts, it creates a symbol for the name export:

```typescript
    function bindExportDeclaration(node: ExportDeclaration) {
        if (!container.symbol || !container.symbol.exports) {
            // Export * in some sort of block construct
            bindAnonymousDeclaration(node, SymbolFlags.ExportStar, getDeclarationName(node)!);
        }
        else if (!node.exportClause) {
            // All export * declarations are collected in an __export symbol
            declareSymbol(container.symbol.exports, container.symbol, node, SymbolFlags.ExportStar, SymbolFlags.None);
        }
        else if (isNamespaceExport(node.exportClause)) {
            // declareSymbol walks up parents to find name text, parent _must_ be set
            // but won't be set by the normal binder walk until `bindChildren` later on.
            setParent(node.exportClause, node);
            declareSymbol(container.symbol.exports, container.symbol, node.exportClause, SymbolFlags.Alias, SymbolFlags.AliasExcludes);
        }
    }
```

**Comment by Boshen** [comment]
I checked other ASTs, although the spec says it's an `IdentiferName`, we need to change it to a `IdentifierReference`

```
[ModuleExportName](https://tc39.es/ecma262/#prod-ModuleExportName) :
  [IdentifierName](https://tc39.es/ecma262/#prod-IdentifierName)
  [StringLiteral](https://tc39.es/ecma262/#prod-StringLiteral)
```
```rust
pub enum ModuleExportName<'a> {
    Identifier(IdentifierName<'a>), // << change back to identifier reference
    StringLiteral(StringLiteral<'a>),
}
```

**Comment by Dunqing** [comment]
I'm glad you're considering changing it to IdentifierReference, I totally agree!

**Comment by Boshen** [comment]
I'm working on this.
